src/symbols.c

```diff
@@ -1147,21 +1126,46 @@ static void parents_table_tree_value_free(gpointer data)
 
 
 /* adds a new element in the parent table if its key is known. */
-static void update_parents_table(GHashTable *table, const TMTag *tag, const gchar *parent_name,
-		const GtkTreeIter *iter)
+static void update_parents_table(GHashTable *table, const TMTag *tag, const GtkTreeIter *iter)
 {
+	const gchar *name;
+	gchar *name_free = NULL;
 	GTree *tree;
-	if (g_hash_table_lookup_extended(table, tag->name, NULL, (gpointer *) &tree) &&
-		! utils_str_equal(parent_name, tag->name) /* prevent Foo::Foo from making parent = child */)
+
+	if (EMPTY(tag->scope))
+	{
+		/* simple case, just use the tag name */
+		name = tag->name;
```

Comment:
What about doing g_strdup(tag->name) here and once more below and completely eliminating the name_free variable?

Reply:
That was a vain attempt at over-optimization to avoid memory allocation in the simple case.  And it'd need profiling, but it might be worth it as it's likely to be the most common case (root scope symbol and no duplicate).  But if it just adds complexity for the reader I'll drop it.

Reply:
Ah, OK, makes sense then. It was just a suggestion, it can stay the way it is.